### ISSUE TITLE
Query Syntax: where condition is `false` but it is included in the search results

### ISSUE THREAD

**Issue by HanJaeJoon** [datalust/seq-tickets#2044]
**Describe the bug**
The where condition using `@Timestamp` is `false` but is included in the search results.

**To Reproduce**
Search using the query below
```
select @Timestamp, @Timestamp < DateTime('2023-12-13T02:02:51.2622192Z'), ToIsoString(@Timestamp)
from stream
where true
and @Timestamp > DateTime('2023-11-15T03:55:01.0000000Z')
and @Timestamp < DateTime('2023-12-15T03:55:01.0000000Z')
and @Timestamp < DateTime('2023-12-13T02:02:51.2622192Z')
order by @Timestamp desc
```

**Expected behavior**
when if the where condition is `false`, it should be excluded from the search results.

**Screenshots**
![image](https://github.com/datalust/seq-tickets/assets/31728365/d8162360-7463-4b61-a426-3eb008fcbbf6)

**Environment:**
 - OS: Windows 11
 - Browser Chrome 120.0.6099.71
 - Seq Version 2023.4.10219
 - Using Docker? No

**Additional context**

**Comment by KodrAus** [comment]
Hi @HanJaeJoon :wave:


Thanks for the report! This is indeed a bug. If you remove the `DateTime('2023-12-15T03:55:01.0000000Z')` clause then it should produce the result you expect.

**Comment by HanJaeJoon** [comment]
@KodrAus Thanks for your quick response.
It works. But is there any way to make it searchable while keeping the current query?

**Comment by KodrAus** [comment]
@HanJaeJoon It's unfortunately a bug in our underlying query engine, so we'll get it patched up, but I think removing that clause is the current workaround.

**Comment by KodrAus** [comment]
We've patched this up. I'll drop another note on this issue once we've got a build deployed with the fix. At this stage it's part of the 2024.1 preview.

Thanks again for the report @HanJaeJoon!

**Comment by KodrAus** [comment]
This is fixed in the latest 2024.1.10873-pre release.
